keps/sig-auth/20190607-certificates-api.md

```diff
@@ -132,6 +134,85 @@ A typical successful issuance proceeds as follows.
    the `.Status.Certificate` field.
 1. The requestor observes the update, and stores the certificate locally.
 
+### Signers
+CSRs have a `signerName` field which is used to specify which signer the CSR creator wants to sign the certificate.
+To support migration from v1beta1 to v1, this required field will be defaulted in v1beta1 (optional in openapi), but
+not defaulted and required in v1 :
+ 1. If it's a kubelet client certificate, it is assigned "kubernetes.io/kubelet-client".
+ 2. Otherwise, it is assigned "kubernetes.io/legacy-unknown".
+
+There will be field selector support to make approvers and signers easier to write.
+
+All signers should provide information about how they work so that clients can predict what will happen to their CSRs.
```

Comment:
Is this a programmatic API or just some text in the docs somewhere?

If it's a programmatic API, what's the use case? Could you provide an example of a program changing its behavior after querying the API? I'm wary of bolting this on without a practical need.

If it's docs, why would k8s prescribe the format? Is there a central curated repository for `signerName`s?

Reply:
Just docs.

> If it's docs, why would k8s prescribe the format?

This isn't prescribing the format, just the aspects which should be described.

> Is there a central curated repository for signerNames?

I would only anticipate hosting documentation for behavior of `kubernetes.io/...` signerNames

Reply:
Should this section say "All predefined kubernetes.io/... signers should provide information..."?
When someone decides to build a new custom signer, there's no expectation of them providing the same description.

Reply:
> When someone decides to build a new custom signer, there's no expectation of them providing the same description.

All signers *should* define this information, since they are effectively defining an API.

Reply:
But there's no way to enforce it without embedding the info in the actual API. Maybe the wording should be `strongly encouraged` instead of `should`?

Reply:
I think should is exactly what we mean (in the RFC sense)